src/server/auth.ts

```diff
@@ -53,3 +55,43 @@ export default async function initializePatronTokenAuth(req: NextRequest) {
 
   return patronTokenResponse
 }
+
+/**
+ * Creates and returns redirect url from initializePatronTokenAuth() response
+ */
+export function getLoginRedirect(req) {
+  if (
+    !req.patronTokenResponse ||
+    !req.patronTokenResponse.isTokenValid ||
+    !req.patronTokenResponse.decodedPatron ||
+    !req.patronTokenResponse.decodedPatron.sub
+  ) {
```

Comment:
This works but it _shouldn't_ work. In DFE, we set the API patron response into the Express `req` object. We can't and shouldn't do that in the Next.js `req` object. In this case `req.patronTokenResponse` will always return an empty object since it doesn't exist.

This works because in the /pages/account/index.tsx file, it is only called when a patron is not authenticated. So if they're not authenticated and because `req.patronTokenResponse` is always empty, this function will always return the url. I think that's fine and how this function should work.

Instead of returning the `redirect` string or undefined, always return the redirect url, so you can remove the logic check. This is also under the assumption that this function should only be called when a user is authenticated (add that to the function comment block).

Another approach is to pass the `isAuthenticated` value to this function and then return the string only when it's false. But I think the way you set it up where you call this function when `isAuthenticated` is already false makes more sense.